docs/framework-overview.md

```diff
@@ -0,0 +1,398 @@
+regular software installation with the `eb` command:
+
+```mermaid
+stateDiagram-v2
+    easybuild.main --> tools
+    robot --> build_install
+    build_install --> easyblock
+    
+    state "easybuild.tools" as tools{
+        [*] --> options
+        options --> parse_ec
+        process_ec --> robot
+        
+        state "easybuild.tools.options
+        • parsing of options
+        • set_up_configuration( )" as options
+        
+        state "easybuild.tools.robot
+        • resolve_dependencies( )" as robot
+    }
+
+    state "build_and_install_software()" as build_install
+    
+    state "easybuild.framework.easyblock" as easyblock {
+        [*] --> build_install_one
+        build_install_one --> get_easyblock
+        get_easyblock --> easyblock_class
+
+        state "easybuild.framework.easyblock
+        • build_and_install_one()" as build_install_one
+        
+        state "class EasyBlock
+        • run_all_steps()" as easyblock_class
+    }
+
+    state "easybuild.framework.easyconfig" as easyconfig{
+        parse_ec --> process_ec
+
+        state "easybuild.framework.easyconfig.tools
+        • parse_easyconfigs()" as parse_ec
+
+        state "easybuild.framework.easyconfig.easyconfig
+        • process_easyconfig()" as process_ec
+
+        state "easybuild.framework.easyconfig.easyconfig
+        • get_easyblock_class()" as get_easyblock
+    }
+    
+```
+
+The process starts with `easybuild.tools` taking care of configuration options
+passed to EasyBuild. Either through the command line, configuration files or in
+the environment.
+
+The scope of the EasyBuild execution is ultimately defined by the target
+easyconfig files given in the `eb` command. They will define the number of
+actions to carry out, such as package installations, fetching of sources or
+opening of PRs. Therefore, target easyconfig files are parsed early in the
+process, just after the runtime options have been resolved. The tooling to
+parse `EasyConfig` files is located in a specific `tools` module in
+`easybuild.framework.easyconfig.tools`, while the `EasyConfig` class is found
+in `easybuild.framework.easyconfig.easyconfig`.
+
+Once the easyconfig files are properly parsed, a common action is to resolve
+the package dependencies for those installations with the
+[`--robot`][use_robot] option. This is handled in `easybuild.tools.robot` which
+provides methods to search easyconfig files, resolve dependencies and check
+conflicts between packages.
+
+The *robot* step returns an ordered list of all software packages that have to
+be installed to fulfill the dependency requirements of the requested
+installation (or throws an error if those dependencies cannot be fulfilled).
+EasyBuild will iterate over that list carrying out each installation in
+sequence.
+
+The installation procedure of each software package is defined by the
+[EasyBlock][easyblocks] attached to it. This means that EasyBuild creates and
```

Comment:
Same here, only use `EasyBlock` if you're referring to the class, just use "easyblock" when refer to the "thing".

Also, typo at the end (`and` -> `an`)

```suggestion
[easyblock][easyblocks] attached to it. This means that EasyBuild creates an